### DISCUSSION TITLE
Change user credentials for a different MS user.

### DISCUSSION THREAD

**Discussion by ghysner** [abraunegg/onedrive#2207] [Q&A] [answered]
I need to change my onedrive client to use a different MS user account. How do I change the MS URI stored on my machine?

**Comment by abraunegg** [comment, marked as answer]
@ghysner80 
> I need to change my onedrive client to use a different MS user account. How do I change the MS URI stored on my machine?

Its not really clear here what you are trying to do.

To use a different account, all you need to do is to call `--logout` and re-authenticate with the new credentials - however, this will still be using your existing configured 'sync_dir' .. thus all your existing data locally, will then be uploaded | merged into the new account ..... this might not be desirable for you. Suggest that you read https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-to-use-multiple-onedrive-accounts--configurations on how to configure a separate account & directory.

A simpler alternative here would be to rename your existing 'sync_dir' (typically `~/OneDrive/`) to something else .. so that your old data is preserved elsewhere locally.

> How do I change the MS URI stored on my machine?

This part is the unclear piece .. what do you mean by this? The authentication response or the URL that the authentication is requested from?

If the latter, you may need to configure your application tenant ... please see: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#configuring-the-client-for-single-tenant-application-use

**Comment by ghysner** [comment]
Thanks for you quick response.

I have onedrive on my pc. I do not want to use the credentials of my local
user for my remote headless linux box. I wish to sign in to MS and get
different credentials for onedrive. After entering the RLI on the remote
onedrive app, when I do a dry-run I still get my local PC's onedrive files.

**Comment by abraunegg** [comment]
@ghysner80 
> I have onedrive on my pc. I do not want to use the credentials of my local
user for my remote headless linux box. I wish to sign in to MS and get
different credentials for onedrive. After entering the RLI on the remote
onedrive app, when I do a dry-run I still get my local PC's onedrive files.

Essentially what you are seeing is correct.

When you login to OneDrive your your PC using `user@your.account.with.microsoft` - and you login online via a Web Browser or using say the Windows app or Mac app or whatever - you will see your files / your data.

If you then use, that *same* account -  `user@your.account.with.microsoft` on a remote PC, using OneDrive Client for Linux - you *will* see your data from your PC & what you have stored with Microsoft online. This is how OneDrive works.

From the sounds of it you want to create a second account with Microsoft OneDrive for your remote PC - `some.other.user@some.other.account.with.microsoft` - this way, when using this account with OneDrive Client for Linux, you will not see your local PC files as it is 100% a separate OneDrive account.
